### PR TITLE
Storage integration test on downloading and grouping disputed territories.

### PR DESCRIPTION
**Pull request by bykoianko**
1. Рефакторинг storage_integration_tests;
2. Добавление теста на загрузку спорных территорий;

@syershov PTAL

### CODE REVIEW

File: storage/storage_integration_tests/test_defines.cpp [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#include "storage/storage_integration_tests/test_defines.hpp"
+
+string const kMapTestDir = string("map-tests");
```
Comment (ygorshenin) [inline storage/storage_integration_tests/test_defines.cpp]:
```c++
namespace storage
{
string const kMapTestDir("map-tests");
string const kTestWebServer("http://...");
}  // namespace storage
```

File: storage/storage_integration_tests/storage_http_tests.cpp [hunk 2]
```diff
@@ -97,3 +122,60 @@ UNIT_TEST(StorageDownloadNodeAndDeleteNodeTests)
   storage.DeleteNode(kCountryId);
   TEST(!platform.IsFileExistsByFullPath(mwmFullPath), ());
 }
+
+UNIT_TEST(StorageDownloadAndDeleteDisputedNodeTests)
+{
+  WritableDirChanger writableDirChanger(kMapTestDir);
```
Comment (ygorshenin) [inline storage/storage_integration_tests/storage_http_tests.cpp]: A lot of code here uses following pattern:
```c++
WritableDirChanger writableDirChanger(...);
Storage storage(...);
... test code ...
```

Comment (ygorshenin) [inline storage/storage_integration_tests/storage_http_tests.cpp]: I think it's time to use UNIT_CLASS_TEST.

File: storage/storage_integration_tests/test_defines.cpp (hunk 1 above)
Comment (bykoianko) [inline storage/storage_integration_tests/test_defines.cpp]: Done

File: storage/storage_integration_tests/storage_http_tests.cpp (hunk 2 above)
Comment (bykoianko) [inline storage/storage_integration_tests/storage_http_tests.cpp]: Yes. Let's start form this place. Now it's good time for it.

Comment (bykoianko) [inline storage/storage_integration_tests/storage_http_tests.cpp]: DONE

**Comment by bykoianko** [comment]
@syershov @ygorshenin PTAL

File: storage/storage_integration_tests/storage_http_tests.cpp [hunk 3]
```diff
@@ -51,49 +70,119 @@ UNIT_TEST(StorageDownloadNodeAndDeleteNodeTests)
...
-  Platform & platform = GetPlatform();
+  InitStorage(GetStorage(), Update, progressFunction);
+
+  string const mwmFullPath = GetMwmFullPath(kCountryId, GetVersion());
+  string const downloadingFullPath = GetDownloadingFullPath(kCountryId, GetVersion());
+  string const resumeFullPath = GetResumeFullPath(kCountryId, GetVersion());
 
   // Downloading to an empty directory.
-  storage.DownloadNode(kCountryId);
+  GetStorage().DownloadNode(kCountryId);
```
Comment (ygorshenin) [inline storage/storage_integration_tests/storage_http_tests.cpp]: You don't need GetStorage()/SetStorage() API. Just make m_storage protected and use it here as is.

Comment (bykoianko) [inline storage/storage_integration_tests/storage_http_tests.cpp]: Done

**Comment by bykoianko** [comment]
DUDE JTTP

**Comment by bykoianko** [comment]
@ygorshenin PTAL

File: storage/storage_integration_tests/storage_http_tests.cpp [hunk 4]
```diff
@@ -51,49 +70,119 @@ UNIT_TEST(StorageDownloadNodeAndDeleteNodeTests)
...
+public:
+  StorageHttpTest()
+    : m_writableDirChanger(kMapTestDir), m_storage(COUNTRIES_FILE),
+      m_version(strings::to_string(m_storage.GetCurrentDataVersion())),
+      m_cleanupVersionDir(m_version)
+  {
+    TEST(version::IsSingleMwm(m_storage.GetCurrentDataVersion()), ());
+  }
+
+  string const & GetVersion() const { return m_version; }
```
Comment (ygorshenin) [inline storage/storage_integration_tests/storage_http_tests.cpp]: Этот метод тоже не нужен, так?

Comment (ygorshenin) [inline storage/storage_integration_tests/storage_http_tests.cpp]: Его можно заменить на прямое использование m_version.

Comment (bykoianko) [inline storage/storage_integration_tests/storage_http_tests.cpp]: Готово

**Comment by bykoianko** [comment]
@ygorshenin PTAL

**Comment by bykoianko** [comment]
@ygorshenin PTAL

**Comment by ygorshenin** [comment]
LGTM

**Pull request merged**